### ISSUE TITLE
Doesn't support HTTPS on OSX/Windows/iOS

### ISSUE THREAD

**Issue by polachok** [Metaswitch/swagger-rs#43]
Mac OS X uses `security_framework` as backend in `native_tls`, so there is no `openssl` in backend:

```
 Compiling swagger v1.0.1 (file:///Users/plhk/Documents/src/swagger-rs)
error[E0433]: failed to resolve. Could not find `openssl` in `backend`
  --> src/connector.rs:39:34
   |
39 |             native_tls::backend::openssl::TlsConnectorBuilderExt::from_openssl(ssl);
   |                                  ^^^^^^^ Could not find `openssl` in `backend`

error[E0433]: failed to resolve. Could not find `openssl` in `backend`
  --> src/connector.rs:83:34
   |
83 |             native_tls::backend::openssl::TlsConnectorBuilderExt::from_openssl(ssl);
   |                                  ^^^^^^^ Could not find `openssl` in `backend`

```
I guess the same error would happen on Windows.

**Comment by bjgill** [comment]
Thanks for the issue.

I don't immediately know how to fix (and don't have the time to do so). However, a quick workaround would be to use `#[cfg(target_os = "linux")]` to swap out the offending functions for `unimplemented!`. That'd give you something that compiled, but then panicked if it attempts to use HTTPS.

I'll have a go at the workaround later today.

**Comment by bjgill** [comment]
You've started work on this before I could get there. Unassigning myself on that basis.

**Comment by bjgill** [comment]
Workaround shipped in [v1.0.2](https://crates.io/crates/swagger/1.0.2). Thanks for the fix. I'm going to leave this issue open because the underlying problem remains unfixed.

**Comment by richardwhiuk** [comment]
https://github.com/Metaswitch/swagger-rs/pull/79/files tweaks the workaround so the relevant functions are missing, rather than panic.

**Comment by richardwhiuk** [comment]
#79 does better than I stated above, and brings HTTPS support for OS X, Windows and iOS.

**Comment by richardwhiuk** [comment]
This is fixed in Swagger 4.0, which is now available on crates.io
